x509/src/attr.rs

```diff
@@ -87,3 +91,164 @@ pub struct AttributeTypeAndValue<'a> {
     /// Value of the attribute
     pub value: Any<'a>,
 }
+
+#[derive(Copy, Clone)]
+enum Escape {
+    None,
+    Some,
+    Hex(u8),
+}
+
+struct Parser {
+    state: Escape,
+    bytes: Vec<u8>,
+}
+
+impl Parser {
+    pub fn new() -> Self {
+        Self {
+            state: Escape::None,
+            bytes: Vec::new(),
+        }
+    }
+
+    fn push(&mut self, c: u8) {
+        self.state = Escape::None;
+        self.bytes.push(c);
+    }
+
+    pub fn add(&mut self, c: u8) -> Result<(), der::Error> {
+        match (self.state, c) {
+            (Escape::Hex(p), b'0'..=b'9') => self.push(p | (c - b'0')),
+            (Escape::Hex(p), b'a'..=b'f') => self.push(p | (c - b'a' + 10)),
+            (Escape::Hex(p), b'A'..=b'F') => self.push(p | (c - b'A' + 10)),
+
+            (Escape::Some, b'0'..=b'9') => self.state = Escape::Hex((c - b'0') << 4),
+            (Escape::Some, b'a'..=b'f') => self.state = Escape::Hex((c - b'a' + 10) << 4),
+            (Escape::Some, b'A'..=b'F') => self.state = Escape::Hex((c - b'A' + 10) << 4),
+
+            (Escape::Some, b' ' | b'"' | b'#' | b'=' | b'\\') => self.push(c),
+            (Escape::Some, b'+' | b',' | b';' | b'<' | b'>') => self.push(c),
+
+            (Escape::None, b'\\') => self.state = Escape::Some,
+            (Escape::None, ..) => self.push(c),
+
+            _ => return Err(ErrorKind::Failed.into()),
+        }
+
+        Ok(())
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.bytes
+    }
+}
+
+impl<'a> AttributeTypeAndValue<'a> {
```

Comment:
It doesn't look like you're using the `'a` lifetime anywhere, in which case it can be omitted:

```suggestion
impl AttributeTypeAndValue<'_> {
```

...and so on below